hw/nfc-nci.c

```diff
@@ -662,28 +663,20 @@ nfc_delivery_deactivate_cmd_cb(void* data, union nci_packet* pkt)
     assert(data);
 
     param = (struct nfc_deactivate_ntf_param*)data;
-    res = nfc_create_deactivate_ntf(param->type, param->reason, pkt);
+    res = nfc_create_deactivate_ntf(nfc, param->type, param->reason, pkt);
```

Comment:
AFAICT this line is the only reason why you need the 'nfc' parameter in the delivery callback. Please pass this parameter as parameter field in |struct nfc_deactivate_ntf_param|, and remove the additional parameter from the delivery callback and related code.